### ISSUE TITLE
Jira Service Management Connector

### ISSUE THREAD

**Issue by Weves** [danswer-ai/danswer#2281]
Pull in all tickets from a specified [Jira Service Management](https://www.atlassian.com/software/jira/service-management) project. 

Checkout the connector creation README here for more details on the best way to add new connectors: https://github.com/danswer-ai/danswer/blob/main/backend/danswer/connectors/README.md.

**Comment by Weves** [comment]
Note on the above: please check the `Before opening PR` section for additional requirements (testing + style)!

**Comment by Mubashirshariq** [comment]
@Weves I had one doubt i found jira-danswer in backend/danswer/connectors/danswer-jira, what needs to be implemented if it is already there?

**Comment by Weves** [comment]
@Mubashirshariq we have a Jira connector already, but it does not work for Jira Service Management specifically (only traditional Jira). We'd like a separate JSM connector. It likely will be similar to the existing Jira connector.

**Comment by Weves** [comment]
@jaykayudo, thanks for the Pull Request! Added a few comments

**Comment by 0Anshu1** [comment]
@Weves I need to submit only one type of connector(any one out of load,poll or event) or can I submit all three ?

**Comment by harshavardhangsv** [comment]
/attempt [#2281](https://github.com/danswer-ai/danswer/issues/2281)

**Comment by harshavardhangsv** [comment]
@Weves I have raised a pr for this. can you please review it.

**Comment by voarsh2** [comment]
I was hoping you'd implement for DC as well - esp given Atlassian Cloud already has Rovo

**Comment by harshavardhangsv** [comment]
Hey @Weves , did you get a chance to review the pr I have raised for this

**Comment by serunkuma** [comment]
/attempt [#2281](https://github.com/onyx-dot-app/onyx/issues/2281)

**Comment by stdthoth** [comment]
@Weves is help still needed on this ??

**Comment by techsouvik** [comment]
is this issue open?

**Comment by aditi-dsi** [comment]
@yuhongsun96 I'd like to work on this issue, could you please assign this to me?

**Comment by Excellencedev** [comment]
unstale

**Comment by Excellencedev** [comment]
@Weves Please assign me to this

**Comment by nifski** [comment]
Hi @Weves!

I noticed the previous attempt (PR #5392) has been stale for months with merge conflicts and failing tests.

I have picked this up and:

Ported the logic to a fresh branch to resolve the conflicts.

Fixed the Icon format issue (SVG vs PNG) that was causing the build bot to fail.
Verified the JSM connector configuration matches the new frontend schema.
I am finalising the tests now and will open a clean PR shortly to close this out.

**Comment by matiasmagni** [comment]
/attempt #2281 

# Implementation Plan: Jira Service Management Connector

## Issue
[#2281](https://github.com/onyx-dot-app/onyx/issues/2281) - Jira Service Management Connector

## Objective
Create a new connector that pulls all tickets from a specified Jira Service Management (JSM) project. JSM is a specialized version of Jira for IT service management, customer support, and operations. While it uses the same underlying Jira REST API, it has specific project types and issue types (Incident, Request, Problem, Change, etc.).

## Background

### What is Jira Service Management?
- JSM is an extension of Jira designed for IT service management (ITSM) and customer support
- Uses the same REST API as regular Jira, but with service management-specific features
- Projects are configured as "Service Management" projects
- Issue types include: Incident, Service Request, Problem, Change, etc.
- May have customer-facing portals for self-service

### Existing Implementation
- There is already a fully-featured `JiraConnector` that:
  - Supports both Jira Cloud (v3 API) and Server/Data Center (v2 API)
  - Implements `CheckpointedConnectorWithPermSync` and `SlimConnectorWithPermSync`
  - Uses JQL queries to fetch issues
  - Processes issues into documents with comprehensive metadata
  - Located at: `backend/onyx/connectors/jira/connector.py`

### Approach
Since JSM uses the same Jira API, we can leverage the existing Jira connector infrastructure while creating a specialized connector that:
1. Targets JSM projects specifically
2. May include JSM-specific metadata (e.g., customer portal information, SLA data)
3. Has a distinct `DocumentSource` enum value for proper categorization

## Implementation Steps

### Phase 1: Backend Core Implementation

#### 1.1 Add DocumentSource Enum Value
**File**: `backend/onyx/configs/constants.py`

- [ ] Add `JIRA_SERVICE_MANAGEMENT = "jira_service_management"` to the `DocumentSource` enum (around line 178, after `JIRA`)

**Example:**
```python
JIRA = "jira"
JIRA_SERVICE_MANAGEMENT = "jira_service_management"  # New entry
SLAB = "slab"
```

#### 1.2 Create JSM Connector Module
**Directory**: `backend/onyx/connectors/jira_service_management/`

Create the following files:
- `__init__.py` - Module initialization
- `connector.py` - Main connector implementation
- `utils.py` - JSM-specific utilities (if needed)

**Implementation Strategy:**
- [ ] Create `JiraServiceManagementConnector` class that extends or reuses the existing `JiraConnector` logic
- [ ] Key differences from regular Jira connector:
  - Constructor should accept a `jsm_project_key` parameter (JSM project key)
  - JQL query should filter for the specific JSM project: `project = "PROJECT_KEY"`
  - Optionally filter by JSM-specific issue types (Incident, Request, Problem, Change)
  - May add JSM-specific metadata fields (SLA information, customer portal links, etc.)
- [ ] Inherit from `CheckpointedConnectorWithPermSync` and `SlimConnectorWithPermSync` (same as JiraConnector)
- [ ] Reuse existing utilities from `backend/onyx/connectors/jira/utils.py` and `backend/onyx/connectors/jira/access.py`

**Key Implementation Details:**
```python
from onyx.connectors.jira.connector import (
    JiraConnectorCheckpoint,
    _perform_jql_search,
    process_jira_issue,
    # ... other reusable components
)
from onyx.connectors.interfaces import CheckpointedConnectorWithPermSync, SlimConnectorWithPermSync
from onyx.configs.constants import DocumentSource

class JiraServiceManagementConnector(
    CheckpointedConnectorWithPermSync[JiraConnectorCheckpoint],
    SlimConnectorWithPermSync,
):
    def __init__(
        self,
        jira_base_url: str,
        jsm_project_key: str,  # Specific to JSM projects
        comment_email_blacklist: list[str] | None = None,
        batch_size: int = INDEX_BATCH_SIZE,
        labels_to_skip: list[str] = JIRA_CONNECTOR_LABELS_TO_SKIP,
        # Optional: filter by JSM issue types
        include_issue_types: list[str] | None = None,  # e.g., ["Incident", "Service Request"]
    ) -> None:
        # Initialize similar to JiraConnector but with JSM-specific config
```

#### 1.3 Register Connector in Registry
**File**: `backend/onyx/connectors/registry.py`

- [ ] Add mapping entry to `CONNECTOR_CLASS_MAP`:
```python
DocumentSource.JIRA_SERVICE_MANAGEMENT: ConnectorMapping(
    module_path="onyx.connectors.jira_service_management.connector",
    class_name="JiraServiceManagementConnector",
),
```

#### 1.4 Update Factory (if needed)
**File**: `backend/onyx/connectors/factory.py`

- [ ] Verify that the existing factory logic works with the new connector (should work automatically via registry)

### Phase 2: Frontend Integration

#### 2.1 Add Source Metadata
**File**: `web/src/lib/sources.ts`

- [ ] Add `jira_service_management` entry to `SOURCE_METADATA_MAP`:
```typescript
jira_service_management: {
  icon: JiraIcon,  // Reuse Jira icon or create JSM-specific icon
  displayName: "Jira Service Management",
  category: SourceCategory.TicketingAndTaskManagement,
  docs: `${DOCS_ADMINS_PATH}/connectors/official/jira-service-management`,
  isPopular: false,  // Set to true if this becomes popular
},
```

- [ ] Import `JiraIcon` if not already imported (or create `JiraServiceManagementIcon`)

#### 2.2 Add Connector Configuration Form
**File**: `web/src/lib/connectors/connectors.ts`

- [ ] Add connector configuration to `connectorConfigs` object:
```typescript
jira_service_management: {
  name: "Jira Service Management",
  source: "jira_service_management",
  connectorSpecificConfig: {
    jira_base_url: {
      type: "string",
      required: true,
    },
    jsm_project_key: {
      type: "string",
      required: true,
    },
    comment_email_blacklist: {
      type: "list",
      required: false,
    },
    labels_to_skip: {
      type: "list",
      required: false,
    },
  },
  credentialNames: ["jira_user_email", "jira_api_token"],  // Reuse Jira credentials
},
```

#### 2.3 Create Connector Form Component (if needed)
**File**: `web/src/lib/connectors/[connector-folder]/JiraServiceManagementConnector.tsx` (if custom form needed)

- [ ] Create form component if JSM needs a different UI than regular Jira
- [ ] Otherwise, can reuse existing Jira form component with appropriate field labels

### Phase 3: Testing

#### 3.1 Unit Tests
**Directory**: `backend/tests/unit/onyx/connectors/jira_service_management/`

- [ ] Create `test_jira_service_management_connector.py`
- [ ] Test basic connector functionality:
  - Initialization
  - Credential loading
  - JQL query generation for JSM projects
  - Document processing
  - Checkpoint handling
- [ ] Test JSM-specific features:
  - JSM project filtering
  - Issue type filtering (if implemented)
  - JSM-specific metadata extraction (if implemented)

#### 3.2 Integration/Daily Tests
**Directory**: `backend/tests/daily/connectors/jira_service_management/`

- [ ] Create `test_jira_service_management_basic.py`
- [ ] Test end-to-end document sync:
  - Full document loading from JSM project
  - Polling for recent updates
  - Permission sync (if applicable)
  - Slim document retrieval
- [ ] Add test setup instructions in test file docstring

#### 3.3 External Dependency Tests (if applicable)
**Directory**: `backend/tests/external_dependency_unit/connectors/jira_service_management/`

- [ ] Create tests for permission sync functionality (if reusing Jira permission logic)
- [ ] Create tests for group sync (if applicable)

### Phase 4: Documentation

#### 4.1 Update Documentation Repository
**Repository**: [onyx-dot-app/documentation](https://github.com/onyx-dot-app/documentation)

- [ ] Create new documentation page: `docs/connectors/official/jira-service-management.md`
- [ ] Include:
  - Overview of JSM connector
  - Prerequisites (Jira instance with JSM enabled, project key)
  - Credential setup instructions:
    - How to create API token in Jira
    - Required permissions (read access to JSM project)
  - Configuration guide:
    - Jira base URL
    - JSM project key
    - Optional settings (comment email blacklist, labels to skip, etc.)
  - Screenshots of:
    - Connector configuration form
    - Where to find project key in Jira
    - Where to create API token
  - Troubleshooting section:
    - Common errors and solutions
    - Permission issues
    - Project key not found errors

### Phase 5: Validation & Edge Cases

#### 5.1 Error Handling
- [ ] Test invalid JSM project key
- [ ] Test project key that doesn't exist
- [ ] Test project key for non-JSM project (should still work but may not have JSM features)
- [ ] Test expired credentials
- [ ] Test insufficient permissions
- [ ] Test rate limiting

#### 5.2 Edge Cases
- [ ] Large JSM projects (thousands of tickets)
- [ ] Empty JSM projects
- [ ] Projects with deleted tickets
- [ ] Projects with restricted visibility
- [ ] JSM projects with customer portal integration
- [ ] Handling of JSM-specific fields (SLA, customer information, etc.)

#### 5.3 Compatibility
- [ ] Test with Jira Cloud (most common for JSM)
- [ ] Test with Jira Data Center (if supported)
- [ ] Test with different JSM project configurations
- [ ] Verify compatibility with existing Jira connector (should not interfere)

## Implementation Considerations

### Code Reusability
- **Reuse existing Jira utilities**: The JSM connector should heavily leverage code from `backend/onyx/connectors/jira/`:
  - `utils.py`: Jira client building, URL building, ADF parsing, etc.
  - `access.py`: Permission handling
  - `connector.py`: Checkpoint handling, JQL search logic
  
- **Consider inheritance vs composition**: 
  - Option A: Create `JiraServiceManagementConnector` as a subclass of `JiraConnector`
  - Option B: Create a separate connector that uses shared utilities
  - **Recommendation**: Option B (composition) is preferred to avoid tight coupling and allow independent evolution

### JSM-Specific Features (Future Enhancements)
- Customer portal information
- SLA tracking data
- Customer-facing request types
- JSM-specific workflows
- Asset management integration

### Performance Considerations
- JSM projects can be very large (especially for enterprise customers)
- Use the same checkpointing strategy as Jira connector
- Consider pagination and rate limiting
- Support both v2 and v3 APIs (like Jira connector)

### Security Considerations
- JSM projects may contain sensitive customer information
- Ensure proper permission checking
- Respect email blacklist for comments
- Handle PII appropriately

## Testing Checklist

Before submitting PR:
- [ ] All unit tests pass
- [ ] Daily tests pass with real JSM project
- [ ] Connector can be created via UI
- [ ] Connector successfully syncs documents
- [ ] Permission sync works (if applicable)
- [ ] Slim connector works for pruning
- [ ] Error handling is comprehensive
- [ ] Documentation is complete
- [ ] Code follows Onyx style guidelines
- [ ] No linting errors

## Dependencies

- Existing dependencies (already in requirements):
  - `jira` Python library
  - Existing Jira connector utilities

- No new dependencies should be required (reuses existing Jira infrastructure)

## Timeline Estimate

- **Phase 1 (Backend Core)**: 2-3 days
- **Phase 2 (Frontend)**: 1 day
- **Phase 3 (Testing)**: 2-3 days
- **Phase 4 (Documentation)**: 1 day
- **Phase 5 (Validation)**: 1-2 days

**Total**: ~7-10 days

## Success Criteria

1. ✅ Connector can be configured via Onyx UI
2. ✅ Connector successfully pulls all tickets from a specified JSM project
3. ✅ Documents are properly indexed and searchable
4. ✅ Permission sync works correctly (if enabled)
5. ✅ Connector handles large projects efficiently
6. ✅ Comprehensive test coverage
7. ✅ Documentation is complete and accurate
8. ✅ No conflicts with existing Jira connector

## Open Questions / Decisions Needed

1. **Should we filter by JSM-specific issue types by default?**
   - Recommendation: No, allow all issue types but make it configurable

2. **Should we extract JSM-specific metadata (SLA, customer portal, etc.)?**
   - Recommendation: Start with basic implementation, add JSM-specific metadata in future PR

3. **Should the connector validate that the project is actually a JSM project?**
   - Recommendation: Yes, validate project type if possible via API

4. **Naming convention**: Use `jira_service_management` or `jira_sm` or `jsm`?
   - Recommendation: `jira_service_management` for clarity

5. **Icon**: Reuse Jira icon or create JSM-specific icon?
   - Recommendation: Reuse Jira icon initially, can add JSM-specific icon later

## References

- [Issue #2281](https://github.com/onyx-dot-app/onyx/issues/2281)
- [Connector Creation README](https://github.com/onyx-dot-app/onyx/blob/main/backend/onyx/connectors/README.md)
- [Existing Jira Connector](https://github.com/onyx-dot-app/onyx/blob/main/backend/onyx/connectors/jira/connector.py)
- [Jira REST API Documentation](https://developer.atlassian.com/cloud/jira/platform/rest/v3/)
- [Jira Service Management API](https://developer.atlassian.com/cloud/jira/service-management/rest/)

## Notes

- The existing Jira connector is well-tested and production-ready. This implementation should follow the same patterns and quality standards.
- JSM uses the same API as regular Jira, so most of the complexity is already handled by the existing Jira connector code.
- Consider this as a "specialized Jira connector" rather than a completely new connector - maximum code reuse is key.

**Comment by nifski** [comment]
Hi @Weves, I received the Algora notification that this is marked 'maintainer approved' — thanks!

Is there anything else needed to get this merged?

**Comment by matiasmagni** [comment]
/claim #2281 PR: https://github.com/onyx-dot-app/onyx/pull/7877

**Comment by ZoneTwelve** [comment]
> Pull in all tickets from a specified [Jira Service Management](https://www.atlassian.com/software/jira/service-management) project.
> 
> Checkout the connector creation README here for more details on the best way to add new connectors: https://github.com/danswer-ai/danswer/blob/main/backend/danswer/connectors/README.md.

I have notice the URL you have provided here is unavailable anymore
It shows `404 - page not found`

Is this the correct link?
https://github.com/onyx-dot-app/onyx/blob/main/backend/onyx/connectors/README.md

**Comment by jefferteleco** [comment]
/attempt #2281

Hi team, I would like to work on this Jira Service Management connector.

My implementation plan:
1. Use the Jira REST API to fetch service requests (tickets) from the Service Management project.
2. Implement the `Connector` interface following the updated structure in `backend/onyx/connectors`.
3. Handle API Token authentication and pagination to retrieve all historical tickets.

**Comment by saimuren** [comment]
I'd like to build this connector. I have experience with both Jira REST API and Python connector patterns.

### Implementation Plan

1. **Connector class**: `JiraServiceManagementConnector` extending the base connector
2. **Authentication**: Support both API token and OAuth 2.0
3. **Data sources**:
   - Service requests/tickets
   - Knowledge base articles
   - SLA information
   - Customer data
4. **Indexing**: Pull and index all JSM content for search
5. **Incremental sync**: Track last sync timestamp for efficient updates

### Architecture
```python
class JiraServiceManagementConnector(BaseConnector):
    def __init__(self, jira_url: str, api_token: str, ...):
        ...
    
    def load_credentials(self) -> dict:
        ...
    
    def poll_source(self) -> Generator[Document, None, None]:
        # Fetch service requests, KB articles, etc.
        ...
```

I'll follow the existing connector patterns (e.g., the standard Jira connector) to maintain consistency. ETA: 2-3 days.

**Comment by jefferteleco** [comment]
Hi everyone! Just a quick update that I have already completed this issue. My PR (#8557) is fully functional, has passed all the AI/bot reviews, includes the required demo video, and is currently waiting for the core team to approve the workflows and do the final review. Thanks!

**Comment by justinleeyang** [comment]
I can take bounty #2281. I will share a scoped implementation plan and first PR shortly.

**Comment by jefferteleco** [comment]
Hi @justinleeyang and maintainers! Just a heads up, I actually submitted a Pull Request for this exact bounty a couple of weeks ago. It already implements the full Jira Service Management Connector with auth and error handling. Maintainers, please let me know if you need any adjustments on my pending PR!"

**Comment by VasuBansal7576** [comment]
/attempt #2281

Implementation plan: I'll build the Jira Service Management connector by integrating with the Jira REST API to pull tickets from a specified JSM project. I have experience building similar API connectors (Plaid, SAP, AWS) for an open source project called Hive. Will follow the connector creation README and submit a PR with a demo video.

**Comment by kacperkostrzab** [comment]
Hi, I'm a TypeScript Developer with over 2 years of professional experience in QA Automation and backend integrations. I’ve worked extensively with Jira workflows during my time at Tutore and Tuatara, so I have a deep understanding of Jira Service Management's API structure and its field mapping requirements.

I would like to take on this task and build a robust Jira Service Management Connector for Onyx. My plan is to ensure clean, type-safe implementation and handle edge cases like custom fields and status transitions.

Can you please assign this to me? I'm ready to start development immediately.

**Comment by starmovie12** [comment]
Hi @Weves and team! 👋

Quick update — I've submitted a fully working PR for this bounty: #9182

Instead of duplicating 1,000+ lines of Jira connector code (which caused issues in previous attempts), I took an inheritance-based approach. `JiraServiceManagementConnector` extends `JiraConnector` with just **45 lines** of new logic — all indexing, ADF parsing, pagination, and EE permission sync are inherited from the existing battle-tested implementation.

All automated reviews are green — Greptile scored it **4/5** ("Safe to merge") and all Cubic issues have been resolved across multiple review rounds. ✅

Happy to address any feedback. Looking forward to your review!

**Comment by starmovie12** [comment]
Hi everyone, just an update that I have fully implemented this feature in PR #9182. It uses the inheritance approach to avoid code duplication and has passed both Greptile and Cubic AI reviews. 🚀

**Comment by starmovie12** [comment]
Hi @Weves and team! 👋

Fully working implementation submitted in **PR #9182** — all automated reviews passing with perfect scores:

- ✅ **Greptile: 5/5** — *"Safe to merge; all critical paths are correctly implemented"*
- ✅ **Cubic: All P1/P2 issues resolved** across 3 review rounds

### What makes this implementation different from the 5 prior attempts

| Aspect | Prior PRs | This PR |
|--------|-----------|---------|
| **Approach** | Copy-pasted 1,000+ lines of Jira connector | Inherits from `JiraConnector` (~65 lines) |
| **EE Permissions** | Broken or missing | ✅ Correct prefix (`jira_service_management_<group>`) on both indexing and sync paths |
| **Dead code** | `access.py` duplicates, `.coverage` files, placeholder tests | Zero non-production files |
| **Config contract** | Field name mismatches | Frontend interface matches backend constructor exactly |

### Architecture

`JiraServiceManagementConnector` extends `JiraConnector` with just **two overrides**:

1. `_load_from_checkpoint` — re-tags documents with `DocumentSource.JIRA_SERVICE_MANAGEMENT`
2. `_get_project_permissions` — ensures EE group IDs are prefixed correctly for JSM

All JQL search, ADF parsing, checkpoint pagination, hierarchy node generation, credential validation, and permission sync logic are inherited — zero duplication.

The EE permission fix also makes `get_project_permissions` in `access.py` and `page_access.py` accept an optional `source` parameter, which is **fully backward compatible** with the existing Jira connector.

PR link: https://github.com/onyx-dot-app/onyx/pull/9182

Looking forward to your review! 🙏

**Comment by godlymane** [comment]
I've submitted a clean implementation in PR #9351.

**Approach:** Thin subclass of `JiraConnector` — no code duplication. The JSM connector overrides only the `document_source` property and the constructor parameter name. All indexing, checkpointing, permission syncing, and hierarchy logic is inherited.

I reviewed the feedback on PRs #7008 and #9121 and addressed all the issues raised by reviewers (no copy-paste, correct `ConnectorMapping`, proper enum entry, matching frontend field names, no dead code, modern typing).

**Comment by godlymane** [comment]
Hey @Weves — I've opened PR #9351 for this bounty.

**Approach:** Thin subclass of `JiraConnector` (no code duplication), incorporating feedback from PRs #7008 and #9121:
- `JiraServiceManagementConnector` overrides only `__init__` and `document_source`
- Dedicated JSM permission sync functions (doc_sync + group_sync) with shared helpers
- Permission prefix chain passes `DocumentSource` dynamically (no hardcoded `JIRA` for JSM docs)
- Full frontend form, source metadata, credentials, and auto-sync support

Bot reviews (greptile 4/5, cubic passing) — no outstanding issues. CI workflows need first-time contributor approval to run.

**Comment by devjason41** [comment]
/attempt #2281

Implementation plan:
- extend the existing Jira connector into a dedicated Jira Service Management connector rather than copy-pasting connector logic
- use JSM REST endpoints for service desks, requests, SLAs, approvals, participants, queues, and portal metadata
- keep the existing Jira connector behavior intact while exposing only the extension points needed for JSM
- wire the new connector through backend registry, frontend config/credentials/source metadata, and unit tests
- preserve compatibility with the existing connector architecture and EE permission flow

Active implementation is now in PR #9358 under `devjason41`.
PR #9355 was closed only to correct the contributor identity separation.

**Comment by abubakarsabir924-cell** [comment]
/attempt #2281
Hi Onyx Team! I'm a Python developer and GSoC 2026 contributor for Vorta. I understand the contribution guidelines. I’m not asking for an official assignment yet, but I’ve started working on the Jira Service Management (JSM) connector based on the existing connector framework in your codebase.
My plan includes:
Auth: Implementing OAuth2/Personal Access Token flow.
Extraction: Fetching tickets and comments via Jira REST API.
Mapping: Integrating JSM metadata with Onyx's indexing logic.
I'll submit a Draft PR with a demo video as soon as the core logic is ready.

**Comment by zhaog100** [comment]
Hi! I'd like to work on this bounty. I'll fork and submit a PR. Let me know if there are any specific requirements.

**Comment by lanxevo3** [comment]
/attempt #2281

Implementation plan:
1. New `jira_service_management` connector using Jira REST API v2 with JSM Service Desk API enrichment
2. Fetches tickets via JQL filtered to `projectTypeKey = "service_desk"` (or specific project key)
3. Enriches each ticket with request type, participants, and SLA status via `/rest/servicedeskapi/`
4. Implements `LoadConnector` + `PollConnector` interfaces
5. Includes `validate_connector_settings()`, full metadata (status, priority, assignee, request_type, labels)
6. Separate `jsm_user_email` / `jsm_api_token` credentials for independent configuration
7. Integration tests in `backend/tests/daily/connectors/jira_service_management/`

**Comment by godlymane** [comment]
/attempt #2281

PR: https://github.com/onyx-dot-app/onyx/pull/9351

**Implementation:** Thin subclass of `JiraConnector` — reuses all indexing, checkpointing, and permission-sync logic. Only overrides `document_source` and the constructor parameter name. Includes full frontend config, credential defaults, and dedicated EE permission sync (doc_sync + group_sync).

All bot review feedback has been addressed in follow-up commits.

**Comment by ck3960250** [comment]
Hi maintainers — I’d like to take this on.\n\nProposed implementation plan:\n1) Add a Jira Service Management connector scaffold aligned with existing connector abstractions.\n2) Implement auth + project-scoped fetch for issues/comments/attachments metadata.\n3) Add incremental sync with checkpoint/cursor support.\n4) Normalize key fields (status, assignee, priority, labels, updated_at).\n5) Add retry/backoff + rate-limit handling + integration tests.\n\nI can start immediately and target an initial PR with connector skeleton + auth/fetch path first, then follow up with incremental sync and tests.\n\nIf there are any required constraints for accepted auth mode (OAuth vs API token) or expected field mapping, please let me know and I’ll align up front.

**Comment by ck3960250** [comment]
Friendly follow-up: I can start right away. If you prefer a specific auth mode (API token vs OAuth) or required field mapping for JSM tickets/comments, I’ll follow that exact path to keep review fast.

**Comment by promisingcoder** [comment]
/attempt #2281

JSM connector extending existing JiraConnector with request type + SLA metadata enrichment. 3 tests included.

**Comment by chengyixu** [comment]
/attempt #2281

## Implementation Plan

**Goal:** Implement a Jira Service Management connector following the pattern in `backend/danswer/connectors/README.md`.

### Architecture Overview

Following the existing connector pattern (e.g., Jira Software connector if one exists, or similar issue-tracking connectors):

1. **Connector class** — Create `backend/danswer/connectors/jira_service_management/connector.py` implementing the `BaseConnector` interface with:
   - `load_credentials`: Accept Jira Service Management API token, base URL, and project key(s)
   - `load_from_state`: Full sync — paginate all JSM tickets from specified project(s) using Jira REST API v3
   - `poll_source`: Incremental sync — use `updated >= lastSync` JQL filter to fetch only changed tickets
   - Document extraction: Convert JSM ticket fields (summary, description, comments, status, priority, assignee, labels, custom fields) to Danswer `Document` objects with proper metadata

2. **Authentication**: Support both API token (email + token) and OAuth 2.0 (3LO) auth patterns that JSM uses

3. **Data mapping**:
   - Ticket ID + URL as document identifier
   - Summary + Description as primary content
   - Comments as additional content chunks
   - Status, priority, type as metadata for filtering

4. **Credentials schema** — Add JSM credentials model to backend schema with proper validation

5. **Frontend integration** — Add JSM to the connector list in the frontend with appropriate config fields

6. **Tests** — Unit tests with mocked JSM API responses

### Timeline
Ready to start immediately. Estimated 5-7 days for full implementation with tests.

**GitHub:** chengyixu | Python backend experience, familiar with REST API integrations and the connector pattern.

**Comment by Krishna1186** [comment]
Hi, I've submitted a PR implementing this connector: https://github.com/onyx-dot-app/onyx/pull/9621

It adds a `JiraServiceManagementConnector` that indexes tickets from a specified JSM project using JQL with checkpoint-based incremental sync, and extracts JSM-specific metadata including request type, SLA status, and customer request type. Same credential shape as the existing Jira connector. Happy to make any adjustments based on feedback.

**Comment by CuboYe** [comment]
/claim 2281

Claiming territory for a scoped fix attempt now. If maintainer preference differs, I can adapt quickly.

**Comment by CharlesWong** [comment]
/attempt #2281

Implementation plan: Extend `JiraConnector` (inheriting checkpoint sync, slim docs, permission sync) with JSM-specific enrichment. Key differentiator: **dynamic custom field ID discovery** via `/rest/api/2/field` — no hard-coded `customfield_XXXXX` values. Full backend + frontend + 25 unit tests.

**Comment by Krishna1186** [comment]
Hi! PR https://github.com/onyx-dot-app/onyx/pull/9621 for the Jira Service Management connector is ready for review. All automated review feedback has been addressed (6 rounds of fixes). The connector follows the existing JiraConnector pattern with full checkpoint support, permission sync, and dynamic SLA field discovery. Would appreciate a human review when possible — happy to make any changes needed!

**Comment by Koush98** [comment]
/attempt #2281

Hi, I’d like to take this bounty.

I’ve reviewed the connector architecture and existing implementations.

Proposed approach:
- Implement JiraConnector using LoadConnector + PollConnector
- Use Jira REST API (/rest/api/3/search) to fetch issues from a project
- Support pagination and incremental sync (based on updated timestamp)
- Map Jira issues → Onyx document schema (id, title, status, timestamps, etc.)
- Register connector in DocumentSource and factory
- Add frontend config for credentials (base_url, project_key, email, api_token)
- Include tests under backend/tests/daily/connectors
- Provide an end-to-end demo video showing connector setup + data ingestion

Before I start:
- Should polling be based on `updated` field or another preferred field?
- Any required fields for document mapping beyond standard metadata?

Let me know if this aligns with expectations.

**Comment by Robert0303-C** [comment]
/claim #2281

PR submitted: https://github.com/onyx-dot-app/onyx/pull/9748

This PR fulfills the Jira Service Management connector bounty. Demo video will be added shortly.

**Comment by zax0rz** [comment]
Submitted PR #9816 addressing the issues from #9673. Happy to make any changes needed.

**Comment by VasuBansal7576** [comment]
/claim #2281
PR #9881 is ready for review: https://github.com/onyx-dot-app/onyx/pull/9881
@Weves implemented JiraServiceManagementConnector as a thin subclass of the existing JiraConnector — overrides only 3 hooks (source, source_display_name, jql_issue_type_filter). Zero logic duplication, full frontend registration included.

**Comment by leviskiprop04-hub50** [comment]
Bot collaborate with me

**Comment by Samcodio** [comment]
/attempt 2281

I’ll implement a new Jira Service Management connector following the existing architecture of other connectors
Use search API to retrieve all tickets (proper pagination included)
The tickets will be mapped into the platform’s document format and built into the ingestion pipeline same as other connectors
I’ll handle authentication, errors, and test locally, then include a short demo showing successful result.

**Comment by Sushit-prog** [comment]
/attempt #2281

**Implementation Plan : Jira Service Management Connector**

I'll build a complete JSM connector with the following scope:

**Backend:**
- New connector at `backend/onyx/connectors/jira_service_management/`
- Implements `CheckpointedConnector` for efficient incremental sync
- Uses JSM REST API (`/rest/servicedeskapi/`) — distinct from standard Jira API
- Supports both Jira Cloud (API token + email) and Jira Server (PAT)
- Indexes: tickets, request types, service desk metadata, comments, status, priority, reporter, assignee
- Proper pagination, rate limit handling, and error recovery

**Frontend:**
- Config panel for base URL, project keys, and auth credentials

**Tests:**
- Daily connector tests following existing patterns

**Docs:**
- PR to onyx-dot-app/documentation with setup guide + screenshots

Will include a demo video showing end-to-end connector setup and successful indexing.
ETA: 2–3 days.

**Comment by starmovie12** [comment]
/attempt #2281

Hi @Weves and team! 👋

I have already implemented this and submitted PR #9182 for review. Just dropping the attempt command here to ensure it syncs correctly with the Algora bounty tracking. 

As a quick summary of my implementation in PR #9182:
* **Inheritance over duplication:** The connector is extremely lean. By inheriting directly from the existing `JiraConnector`, there is zero duplication of pagination, ADF parsing, or permission logic.
* **Dynamic SLA Field Discovery:** Instead of hardcoded IDs (which break across different tenant deployments), the connector dynamically discovers SLA fields via regex, supporting both Cloud and Server/DC formats.
* **Production-Ready:** Includes 50+ comprehensive unit tests covering SLA extraction, edge cases, and EE permission prefix fixes.
* **Review Ready:** All automated AI reviews (Greptile & Cubic) have been fully addressed and resolved with high confidence scores.

Looking forward to your feedback on the PR once the CI workflows are approved! 🙏

**Comment by starmovie12** [comment]
/claim #2281

Hi @Weves and the Onyx team! 👋

My PR #9182 implements the Jira Service Management connector and is fully complete, tested, and ready for your review.

**Architecture: inheritance over duplication**
Rather than duplicating the 1,000+ line `JiraConnector`, this implementation subclasses it directly. The connector is only ~45 lines — all pagination, ADF parsing, checkpoint, and permission logic is 100% reused from the battle-tested parent class.

**Key highlights:**
- **Dynamic SLA Field Discovery** — instead of hardcoding tenant-specific IDs like `customfield_10010` (which silently fail on most deployments), SLA fields are discovered dynamically via regex, supporting both Cloud and Server/DC formats out of the box.
- **EE Permission Prefix Fix** — setting `_source = DocumentSource.JIRA_SERVICE_MANAGEMENT` ensures the correct `jira_service_management_` group ID prefix is used for both indexing and permission sync, preventing silent EE access control breakage.
- **URL Validation** — `jira_service_management_doc_sync` validates `jira_base_url` before any network calls, raising `ConnectorValidationError` with a clear message on misconfiguration.
- **50+ Unit Tests** — covering SLA discovery, Cloud/Server DC extraction, breach flags, edge cases, and doc sync URL validation across 7 test classes.

**Review bot status:**
- ✅ Greptile — all feedback resolved (4/5, "safe to merge")
- ✅ Cubic — all P1/P2 issues resolved across 3 review rounds

All four code review comments (test source assertions, `_discover_sla_fields` sentinel return, `_get_request_type` safer fallback, duplicate credential interface) have been addressed in the latest commits.

Looking forward to your feedback. Thank you for your time! 🙏

**Comment by mohammedelazrouri** [comment]
Implementation Plan:
I'll implement a dedicated JSM connector using the specific Service Desk API (/rest/servicedeskapi/) instead of just inheriting from the base Jira connector.

Data Scope: Fetching Requests, Comments, and Attachments specifically from JSM projects.

Architecture: Implementing CheckpointedConnector for incremental syncing (crucial for large service desks).

Validation: Adding a pre-sync check for JSM-specific permissions.

Testing: Unit tests using unittest.mock to ensure zero-cost API simulation.

**Comment by mohammedelazrouri** [comment]
[x] Set up a dedicated Jira Service Management (JSM) testing environment.

[x] Verified API connectivity with /rest/servicedeskapi/request (Success 200).

[x] Mapping JSM-specific fields (Customer Request Types & SLAs) which are missing in the current generic Jira connector.

Implementation is underway. I'll be submitting the PR soon focusing on a high-performance CheckpointedConnector for JSM.

**Comment by shehryarakif** [comment]
/attempt #2281
Step 1: Fork and Clone the Onyx Repository
First, fork the main onyx-dot-app/onyx repository and clone your fork:

bash
git clone https://github.com/<your-username>/onyx.git
cd onyx
git checkout -b feature/jsm-connector
Step 2: Add the New Source Type
Add JIRA_SERVICE_MANAGEMENT to the ValidSources enum. This enum lives in two places: the backend Python code and the frontend TypeScript code.

Backend – backend/onyx/configs/constants.py:

python
class DocumentSource(str, Enum):
    # ... existing sources ...
    JIRA = "jira"
    JIRA_SERVICE_MANAGEMENT = "jira_service_management"
Frontend – web/src/lib/sources.ts:

typescript
export enum ValidSources {
    // ... existing sources ...
    Jira = "jira",
    JiraServiceManagement = "jira_service_management",
}
Step 3: Create the Backend Connector
a. Create the directory and files:

bash
mkdir -p backend/onyx/connectors/jira_service_management
touch backend/onyx/connectors/jira_service_management/__init__.py
touch backend/onyx/connectors/jira_service_management/connector.py
b. Implement the connector class in backend/onyx/connectors/jira_service_management/connector.py:

python
import os
from datetime import datetime
from typing import Any, Iterator, Optional

from jira import JIRA  # Onyx already has this dependency

from onyx.configs.constants import DocumentSource
from onyx.connectors.interfaces import (
    BaseConnector,
    CheckpointedConnector,
    ConnectorCheckpoint,
    ConnectorFailure,
    GenerateDocumentsOutput,
    LoadConnector,
    PollConnector,
    SecondsSinceUnixEpoch,
)
from onyx.connectors.models import Document, Section
from onyx.utils.logger import setup_logger

logger = setup_logger()


class JiraServiceManagementConnector(LoadConnector, PollConnector, CheckpointedConnector):
    """Connector for Jira Service Management projects.

    This connector pulls all tickets from a specified JSM project.
    """

    def __init__(
        self,
        jira_base_url: str,
        project_key: str,
        batch_size: int = 100,
    ) -> None:
        self.jira_base_url = jira_base_url.rstrip("/")
        self.project_key = project_key.upper()
        self.batch_size = batch_size
        self.jira_client: Optional[JIRA] = None

    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
        """Authenticate using email + API token (Basic Auth)."""
        email = credentials.get("jira_user_email")
        api_token = credentials.get("jira_api_token")

        if not email or not api_token:
            raise ValueError("Jira Service Management connector requires email and API token.")

        self.jira_client = JIRA(
            server=self.jira_base_url,
            basic_auth=(email, api_token),
        )
        return None

    def _fetch_issues(
        self, start: SecondsSinceUnixEpoch | None = None, end: SecondsSinceUnixEpoch | None = None
    ) -> Iterator[Document]:
        if not self.jira_client:
            raise RuntimeError("Jira client not initialized. Call load_credentials first.")

        jql_parts = [f"project = {self.project_key}"]

        if start is not None:
            start_date = datetime.fromtimestamp(start).strftime("%Y-%m-%d %H:%M")
            jql_parts.append(f"updated >= '{start_date}'")
        if end is not None:
            end_date = datetime.fromtimestamp(end).strftime("%Y-%m-%d %H:%M")
            jql_parts.append(f"updated < '{end_date}'")

        jql = " AND ".join(jql_parts) + " ORDER BY updated ASC"

        start_at = 0
        while True:
            issues = self.jira_client.search_issues(
                jql,
                startAt=start_at,
                maxResults=self.batch_size,
                fields="*all",
                expand="renderedFields,changelog",
            )

            if not issues:
                break

            for issue in issues:
                yield self._issue_to_document(issue)

            if len(issues) < self.batch_size:
                break
            start_at += len(issues)

    def _issue_to_document(self, issue: Any) -> Document:
        """Convert a Jira issue to an Onyx Document."""
        doc_id = f"jsm:{self.project_key}:{issue.key}"
        title = f"[{issue.key}] {issue.fields.summary}"
        description = issue.fields.description or ""

        content_parts = [
            f"Issue: {issue.key}",
            f"Summary: {issue.fields.summary}",
            f"Type: {issue.fields.issuetype.name}",
            f"Status: {issue.fields.status.name}",
            f"Priority: {getattr(issue.fields.priority, 'name', 'None')}",
            f"Reporter: {getattr(issue.fields.reporter, 'displayName', 'Unassigned')}",
            f"Assignee: {getattr(issue.fields.assignee, 'displayName', 'Unassigned')}",
            f"Created: {issue.fields.created}",
            f"Updated: {issue.fields.updated}",
            f"Description: {description}",
        ]

        for comment in issue.fields.comment.comments:
            content_parts.append(
                f"Comment by {comment.author.displayName} ({comment.created}): {comment.body}"
            )

        text = "\n\n".join(content_parts)

        return Document(
            id=doc_id,
            source=DocumentSource.JIRA_SERVICE_MANAGEMENT,
            semantic_identifier=title,
            sections=[Section(link=issue.permalink(), text=text)],
            metadata={
                "project_key": self.project_key,
                "issue_key": issue.key,
                "issue_type": issue.fields.issuetype.name,
                "status": issue.fields.status.name,
                "priority": getattr(issue.fields.priority, "name", None),
                "created": issue.fields.created,
                "updated": issue.fields.updated,
            },
            doc_updated_at=datetime.fromisoformat(issue.fields.updated.replace("Z", "+00:00")),
        )

    def load_from_state(self) -> GenerateDocumentsOutput:
        """Full historical load."""
        documents = list(self._fetch_issues())
        return documents

    def poll_source(
        self, start: SecondsSinceUnixEpoch, end: SecondsSinceUnixEpoch
    ) -> GenerateDocumentsOutput:
        """Incremental sync based on updated timestamp."""
        return list(self._fetch_issues(start=start, end=end))

    def build_dummy_checkpoint(self) -> ConnectorCheckpoint:
        """Checkpoint support for resume capability."""
        return ConnectorCheckpoint(has_more=False, next_checkpoint=None)

    def validate_checkpoint_json(self, checkpoint_json: str) -> ConnectorCheckpoint:
        return ConnectorCheckpoint.from_json(checkpoint_json)

    def build_checkpoint(
        self, checkpoint: ConnectorCheckpoint, documents: list[Document]
    ) -> ConnectorCheckpoint:
        return checkpoint

    def fetch_checkpointed_documents(
        self, checkpoint: ConnectorCheckpoint
    ) -> GenerateDocumentsOutput:
        """Resume from a checkpoint."""
        # For simplicity, we'll just do a full load; can be enhanced later
        return self.load_from_state()
Step 4: Register the Connector in the Factory
In backend/onyx/connectors/factory.py, add the import and the case:

python
from onyx.connectors.jira_service_management.connector import JiraServiceManagementConnector

# ... inside instantiate_connector function ...
    elif source == DocumentSource.JIRA_SERVICE_MANAGEMENT:
        return JiraServiceManagementConnector(
            jira_base_url=connector_specific_config["jira_base_url"],
            project_key=connector_specific_config["project_key"],
        )
Step 5: Frontend Configuration
In web/src/lib/connectors/connectors.ts, add the configuration:

typescript
import { ValidSources } from "@/lib/sources";

// ... inside connectorConfigs object ...
[ValidSources.JiraServiceManagement]: {
    name: "Jira Service Management",
    description: "Index tickets from a Jira Service Management project",
    fields: [
        {
            name: "jira_base_url",
            label: "Jira Base URL",
            type: "text",
            placeholder: "https://your-domain.atlassian.net",
            required: true,
        },
        {
            name: "project_key",
            label: "Project Key",
            type: "text",
            placeholder: "JSM",
            required: true,
        },
    ],
    credentialFields: [
        {
            name: "jira_user_email",
            label: "Email",
            type: "text",
            required: true,
        },
        {
            name: "jira_api_token",
            label: "API Token",
            type: "password",
            required: true,
        },
    ],
},
Add metadata in web/src/lib/sources.ts:

typescript
[ValidSources.JiraServiceManagement]: {
    displayName: "Jira Service Management",
    category: "Ticketing & Tasks",
    description: "Index support tickets from Jira Service Management",
    icon: "jira", // reuse Jira icon
    docsLink: "https://docs.onyx.app/connectors/jira-service-management",
},
Step 6: Add Tests
Create backend/tests/daily/connectors/jira_service_management/test_connector.py:

python
from unittest.mock import MagicMock, patch

import pytest

from onyx.connectors.jira_service_management.connector import JiraServiceManagementConnector


def test_jira_service_management_connector_initialization():
    connector = JiraServiceManagementConnector(
        jira_base_url="https://test.atlassian.net",
        project_key="TEST",
    )
    assert connector.jira_base_url == "https://test.atlassian.net"
    assert connector.project_key == "TEST"
    assert connector.batch_size == 100


def test_load_credentials():
    connector = JiraServiceManagementConnector(
        jira_base_url="https://test.atlassian.net",
        project_key="TEST",
    )
    credentials = {
        "jira_user_email": "test@example.com",
        "jira_api_token": "test-token",
    }
    with patch("onyx.connectors.jira_service_management.connector.JIRA") as mock_jira:
        connector.load_credentials(credentials)
        mock_jira.assert_called_once_with(
            server="https://test.atlassian.net",
            basic_auth=("test@example.com", "test-token"),
        )
Step 7: Test Locally
You can test the connector standalone using the template from the README:

python
if __name__ == "__main__":
    import os

    connector = JiraServiceManagementConnector(
        jira_base_url=os.environ["JIRA_BASE_URL"],
        project_key=os.environ["JIRA_PROJECT_KEY"],
    )
    connector.load_credentials({
        "jira_user_email": os.environ["JIRA_USER_EMAIL"],
        "jira_api_token": os.environ["JIRA_API_TOKEN"],
    })

    for doc in connector.load_from_state():
        print(f"Loaded: {doc.semantic_identifier}")
Step 8: Run Linters and Tests
bash
# Backend linting
cd backend
ruff check .
mypy onyx/connectors/jira_service_management/

# Run tests
pytest tests/daily/connectors/jira_service_management/
Step 9: Commit and Open a Pull Request
Once everything works, commit your changes and push to your fork, then open a PR against the main Onyx repository.

That's the basic roadmap. You'll be adding a new connector to Onyx's 40+ supported data sources. Let me know if you'd like me to elaborate on any part of the implementation!

/claim #2281

**Comment by dodocroswel** [comment]
/attempt #2281

Implementation plan:
- extend the existing Jira connector instead of duplicating connector logic
- add the narrowest Jira Service Management-specific layer needed for project-scoped ingestion
- wire backend registration, frontend config/credentials, and connector tests in the existing Onyx patterns
- keep the PR scoped, include a short demo video, and use /claim #2281 in the PR body

Clarifications I will verify while implementing:
- whether v1 should include comments only or also attachments
- whether one JSM project per connector instance is the preferred scope
- which metadata fields matter most for retrieval quality beyond title/description/status

**Comment by zhuanxqq** [comment]
Canceling my attempt, too many competing PRs already open.

**Comment by LMDtokyo** [comment]
Hi team! I'd like to attempt this bounty.

My plan:
1. Scaffold a new connector at `backend/onyx/connectors/jira_service_management/` 
   using the existing Jira connector as a reference.
2. Implement `LoadConnector` for bulk import of JSM requests via the 
   `/rest/servicedeskapi/` endpoints.
3. Add `PollConnector` for incremental updates based on ticket updated-at.
4. Handle attachments, comments, custom fields, and request types properly.
5. Register the connector in the UI and add tests.

A few quick questions before I start:
- Should the connector support multiple service desks per instance, or one 
  at a time (configurable)?
- Preferred auth: API token (Basic Auth) or OAuth 2.0?
- Any specific field mappings you'd like prioritized?

ETA: 2–3 days after scope confirmation.

GitHub: https://github.com/LMDtokyo

**Comment by atecnoco-arch** [comment]
Hi @LMDtokyo and team! Just a quick heads up to prevent duplicate effort—I’ve already submitted a fully working implementation for this bounty over at PR #10312. Looking forward to the maintainer's review!

**Comment by atecnoco-arch** [comment]
SYSTEM_FLAG: [STATUS: RESOLVED_PENDING_MERGE] [PR_LINKED_DO_NOT_DUPLICATE: #10312] Actions_Paused.

**Comment by zadokdemo-eng** [comment]
Is this bounty still available? I see a pending claim from March, but I have a Python implementation ready for the JSM SLA fields. I can submit today.

**Comment by thinkflow-hub** [comment]
/attempt #2281
## Implementation Plan for Jira Service Management (JSM) Connector

I am taking this on behalf of **ThinkFLOW HUB**. We've analyzed the existing connector architecture and plan to follow the `PollConnector` pattern for incremental indexing.

**Technical Roadmap:**
1. **Target API:** We will specifically target the JSM-specific REST API (`/rest/servicedeskapi/`) to ensure service requests and customer comments are correctly captured, differentiating from the existing Jira Software connector.
2. 2. **Structure:** Implementation will reside in `backend/onyx/connectors/jira_service_management/`, inheriting from `PollConnector` and `LoadConnector`.
3. 3. **Registry:** We will register the new `JIRA_SERVICE_MANAGEMENT` source in `constants.py` and map it in `registry.py`.
4. 4. **Validation:** Full unit tests with mock API responses and a demo video showing the JSM data searchable in Onyx.
Ready to start.

**Comment by orbitwebsites-cloud** [comment]
/attempt #2281

**Implementation plan:**
Implement Jira Service Management API to fetch tickets from a specified project, following the connector creation guidelines in the README.

Will submit a PR shortly.

**Comment by orbitwebsites-cloud** [comment]
/attempt #2281

**Implementation plan:**
Implement Jira Service Management API to fetch tickets from a specified project, following the connector creation guidelines in the README.

Will submit a PR shortly.

**Comment by vikrantwiz02** [comment]
Hi @Weves, Review it once.

I have just submitted a PR for this issue: #10652 

Key Implementation Details:

- Thin Subclass Architecture: Instead of duplicating logic, I refactored the base JiraConnector to be source-configurable, implementing JSM as a thin subclass. This avoids the technical debt seen in prior attempts.
- JSM Validation: The connector specifically validates for the service_desk project type to prevent misconfiguration.
- Portal Enrichment: Tickets are automatically enriched with Customer Portal URLs for better end-user utility.
- EE Permission Sync: Integrated portal-only customer emails into the external permission sync to ensure proper access control in search.

I have verified the backend logic with daily integration tests (test_jsm_basic.py) and confirmed the frontend registration in a local environment.

/claim #2281

**Comment by devman247** [comment]
Hi, I am interested in solving this issue. Please send me an invitation to join.

**Comment by vikrantwiz02** [comment]
Dead response, no need to put ur effort. No feedback nothing. All of u can see, how long contributions have been done.

**Comment by leviskiprop04-hub50** [comment]
Here I'm

**Comment by jshaofa-ui** [comment]
## 💡 Jira Service Management Connector - Implementation Proposal

I've developed a complete JSM connector implementation for Onyx. Here's what it includes:

### Features
- **Ticket Search**: Full-text search across JSM tickets using JQL
- **Knowledge Base**: Index and search JSM KB articles (Confluence-backed)
- **Customer Requests**: Search customer-facing service desk requests
- **SLA Tracking**: Query SLA compliance and breach data
- **Incremental Sync**: Poll-based incremental document synchronization
- **Onyx Compatible**: Implements the Onyx connector interface

### Architecture


### Key Implementation Details
- Async HTTP client with proper authentication (Basic auth with API token)
- Batched document retrieval (100 docs per batch for Onyx)
- Sync state tracking for incremental updates
- Multi-source support (tickets, KB articles, customer requests)
- Full metadata extraction (status, priority, labels, comments, etc.)

### MVP Code
The complete implementation is available with:
- : Full connector implementation (~500 lines)
- : Minimal dependencies (httpx, pydantic)
- : Configuration and usage guide

### Testing
The connector can be tested with:


I'm ready to create a proper PR following the Connector README guidelines. Would love to contribute this to Onyx!

**Comment by brone1323** [comment]
/attempt #2281

Implementation plan: Thin subclass of JiraConnector since Jira and JSM share REST endpoints. Adds DocumentSource.JIRA_SERVICE_MANAGEMENT, JiraServiceManagementConnector(JiraConnector) overriding _source, registry mapping, frontend descriptors, and mirrors the Jira unit-test layout.

PR: https://github.com/onyx-dot-app/onyx/pull/10858 — 5/5 new tests pass, 32/32 Jira regression tests pass.

**Comment by MilosM348** [comment]
Hi @evan-onyx, @wenxi-onyx, @Bo-Onyx — picking up the JSM connector ask from this thread (originally bountied by @Weves in [August 2024](https://github.com/onyx-dot-app/onyx/issues/2281#issuecomment-2322316167) — I noticed @Weves hasn't been active here for a while and saw you three handle most recent connector merges, so wanted to surface the plan to you).

I've read through the 9 closed PRs for #2281 and the 23 currently open ones. The pattern in the closed ones is: structure looks plausible but no tests, no demo, no real-API validation. I'd like to ship a clean implementation that addresses those gaps before opening a PR. Plan below — happy to adjust scope.

### Scope

A new connector under `backend/onyx/connectors/jira_service_management/` registered as `DocumentSource.JIRA_SERVICE_MANAGEMENT`, separate from the existing `jira/` connector (matching @Weves's [explicit ask](https://github.com/onyx-dot-app/onyx/issues/2281#issuecomment-2323158706) for a separate module rather than folding into core Jira).

### Implementation

- `models.py` — pydantic models normalised from the JSM API: `JsmRequest`, `JsmCustomer`, `JsmRequestType` with reporter, participants, organisation IDs, request type, status, priority, timestamps.
- `utils.py` — Basic-Auth session (email + API token), paged `GET` against `/rest/servicedeskapi/request`, typed exception mapping for 401/403/404, ISO-8601 timestamp parsing for the `+0000`-suffixed format Atlassian returns.
- `connector.py` — `JiraServiceManagementConnector` implementing `LoadConnector` and `PollConnector`. Documents are emitted in `INDEX_BATCH_SIZE` batches with primary/secondary owners populated from the JSM reporter + participants, plus `metadata` with issue_key, request type, status, priority, organisation IDs.

### Tests

`backend/tests/unit/onyx/connectors/jira_service_management/test_connector.py` mocking the JSM HTTP responses with `responses` fixtures captured from a real Atlassian Cloud free trial:

- `to_jsm_request` field-coverage on a real `/request/{key}` response
- pagination stops at `isLastPage`
- 401 → `CredentialExpiredError`, 403 → `InsufficientPermissionsError`, 404 → `ConnectorValidationError`
- `load_from_state` yields in `INDEX_BATCH_SIZE` batches
- `poll_source` window-filters on `currentStatus.statusDate`

### Demo

A short screen recording showing: configure the connector against a JSM trial → first indexing run → a search in the Onyx UI returning a JSM ticket. Will attach to the PR.

### Out of scope (happy to add as follow-ups)

- Permission sync (`SlimConnectorWithPermSync`) — only adding if you'd like it; JSM perms are queryable via `/servicedesk/{id}/customer`.
- Per-request comment threads via `/request/{key}/comment` — easy add but doubles index size on busy desks; want your call before including.
- Frontend connector card placement (multiple recent closed PRs broke this file as it's been refactored several times — would prefer to update it after the backend PR has had a first look).

ETA ~5 days from a thumbs-up. Will only `/claim #2281` once the demo records cleanly.

**Comment by jpbarrosg2805** [comment]
/attempt #2281

Implementation plan: I will implement a dedicated Jira Service Management connector instead of extending the existing Jira connector. The connector will use Atlassian's JSM REST API (`/rest/servicedeskapi`), resolve the configured service-desk project key to a service desk ID, fetch customer requests with pagination, include request comments, normalize reporter/participants/request type/status/priority/organization metadata into Onyx Documents, and support incremental polling by JSM latest activity/status timestamps. I will add backend registry/source wiring, frontend connector metadata/configuration, and focused unit tests for project resolution, pagination, comments, polling filters, and 401/403/404 error mapping. I will only submit `/claim #2281` after tests pass and the PR is ready.

**Comment by aimqwest** [comment]
/attempt I can implement the Jira Service Management connector with tests and docs. I can start immediately.

**Comment by aimqwest** [comment]
/attempt #2281\n\nAIMQWEST implementation plan: add dedicated Jira Service Management connector aligned with connector README requirements, tests, and style checks.

**Comment by louisgiang** [comment]
/attempt #2281

Plan:
- Add JiraServiceManagementConnector following the existing JiraConnector pattern (backend/onyx/connectors/jira/)
- Pull tickets from a specified JSM project, support indexing + permission sync if applicable
- Include unit tests + connector README
- Setup local env first (per docs.onyx.app/quickstart) to record demo video — addressing the blocker that closed PR #2285

ETA: 5-7 days

**Comment by louisgiang** [comment]
> /attempt [#2281](https://github.com/onyx-dot-app/onyx/issues/2281)
> 
> Plan:
> 
> * Add JiraServiceManagementConnector following the existing JiraConnector pattern (backend/onyx/connectors/jira/)
> * Pull tickets from a specified JSM project, support indexing + permission sync if applicable
> * Include unit tests + connector README
> * Setup local env first (per docs.onyx.app/quickstart) to record demo video — addressing the blocker that closed PR [Jira service management connector #2285](https://github.com/onyx-dot-app/onyx/pull/2285)
> 
> ETA: 5-7 days

Hi @Weves,

Before I commit to a Jira Service Management connector implementation, could you clarify the bounty status?

I see PR #9881 (@VasuBansal7576) has been open since April 2026 (~42 days) with no maintainer review yet. Wanted to check:

1. Is the JSM connector still actively wanted, or is the bounty essentially paused?
2. If active — are there gaps in #9881's approach that a new PR should address differently?
3. Does the connector need to be a separate `DocumentSource.JIRA_SERVICE_MANAGEMENT` type, or extend the existing Jira connector with a JSM mode?

Happy to wait for guidance before investing time. Thanks!

**Comment by LittleK-513** [comment]
Hi there! I came across this issue and am interested in working on it. Could you please confirm:

1. What is the bounty amount for this issue?
2. Is the bounty still available / unclaimed?
3. What are the exact claim conditions and requirements?

I have experience with Python connectors and would like to understand the full scope before starting. Thanks!

**Comment by maxi-maxima** [comment]
Hi! I’m interested in taking this on.

I’ve looked through the existing Jira connector and it seems like a Jira Service Management connector could reuse much of the Jira auth/client/document conversion path while adding JSM-specific request/ticket handling.

Before I start implementation, could you confirm the expected MVP scope?

1. Should the connector index JSM requests/tickets only, or also request types, customer/org metadata, comments, attachments, and linked KB articles?
2. Should this be a separate `Jira Service Management` source in the connector registry, or an option/extension on the existing Jira connector?
3. Are there any permission-sync requirements for the first version?

Once scope is confirmed, I can submit a focused PR with tests.

**Comment by arthurus36-alt** [comment]
Please claim the bounty payout for #2281. The implementation has been provided in the merged PR #11228.

**Comment by 64johnlee** [comment]
/attempt #2281

**Implementation plan:**
- New `JiraServiceManagementConnector` subclassing `JiraConnector` — reuses all pagination, checkpointing, and permission-sync logic unchanged
- `DocumentSource.JIRA_SERVICE_MANAGEMENT` stamps all indexed documents as a distinct source type
- Default JQL auto-restricts to `project type = "service_management"` when no project/JQL scope is given
- `validate_connector_settings` raises `ConnectorValidationError` if a non-JSM project key is supplied
- Full frontend: ValidSources enum, sources.ts icon entry, connector config UI (base URL / project key / custom JQL / comment blacklist), credential template, auto-sync support
- Unit tests covering registry mapping, document source stamping, JQL generation, and settings validation
- No changes to the existing `JiraConnector` base class

**Comment by kennedydqz-del** [comment]
/attempt https://github.com/onyx-dot-app/onyx/issues/2281

  1. Register New Data Source

  - Add DocumentSource.JIRA_SERVICE_MANAGEMENT to the enum in backend/onyx/configs/constants.py
  - Register the mapping in backend/onyx/connectors/registry.py

  2. Implement the Connector

  Create backend/onyx/connectors/jira_service_management/connector.py with a JiraServiceManagementConnector class
  extending both CheckpointedConnectorWithPermSync and SlimConnectorWithPermSync.

  Core methods:
  - __init__() — accepts base_url, project_key, email_blacklist, labels_to_skip
  - validate_connector_settings() — verifies the project key exists and is a JSM project
  - load_from_checkpoint() — incremental pull with this flow:
    a. Build JQL: project = "{key}" or project type = "service_management" (fallback)
    b. Paginate through Jira REST API to fetch issues
    c. Extract fields: summary, description, comments, attachments
    d. Return a list of Document objects for Onyx to index
  - retrieve_all_slim_documents() — return lightweight doc list (id + title only)
  - doc_sync() / group_sync() — permission sync, mapping Jira project roles to Onyx access groups

  3. JQL Strategy

  - If user provides a project_key → project = "JSM"
  - If no key → project type = "service_management" (pull all JSM projects)
  - Combine with checkpoint timestamp → append AND updated >= "{checkpoint}" for incremental sync
  - Use the fields parameter to limit returned fields and reduce API load

  4. Frontend Configuration

  - web/src/lib/types.ts — add JIRA_SERVICE_MANAGEMENT enum value
  - web/src/lib/sources.ts — add display name, icon, description
  - web/src/lib/connectors/connectors.tsx — config form with:
    - Base URL (text input)
    - Project Key (text input, optional)
    - API Token (credential selector)
    - Email Blacklist (multiline text)
    - Labels to Skip (multiline text)

  5. Tests

  Follow Onyx's three-tier structure:
  - tests/unit/ — unit tests: mock Jira API, verify JQL generation, data parsing, validation logic, checkpoint handling,
   permission mapping
  - tests/external_dependency_unit/ — external dependency tests: placeholder, need real JSM instance
  - tests/daily/ — daily integration tests: placeholder, verify end-to-end pulling

**Comment by NAEL592** [comment]
/attempt #2281

I will inspect the existing connector framework, add the Jira Service Management connector in the narrowest supported shape, and include focused tests/docs for authentication and ticket/service-desk operations.

**Comment by jamilahmadzai** [comment]
/attempt #2281

Plan: submit a narrow dedicated Jira Service Management connector that reuses the existing Jira indexing path, stamps a separate JSM source, scopes unqualified/custom JQL to service spaces, validates project-key configs up front, wires the admin UI/credentials/auto-sync, and includes focused unit coverage plus a daily real-instance test harness.

**Comment by 64johnlee** [comment]
@Weves — Hi! I've been reviewing this bounty issue and see multiple implementation attempts from May 2026 (PRs #11194, #11223) that haven't been reviewed yet, along with earlier PRs like #9881 (50+ days open) and #10938. 

Before I invest time, could you clarify:

1. **Is the bounty still actively wanted?** The issue is marked "maintainer approved" but no PR has been merged since the issue opened 9 months ago — is this still a priority?

2. **What's blocking the existing PRs?** Are there gaps in the approach (e.g., scope, testing, architecture) that caused them not to move forward? Knowing this would help me avoid the same pitfalls.

3. **Recommended approach?** Should I:
   - Pick up one of the stalled PRs and help fix it?
   - Create a minimal MVP (e.g., extend the existing Jira connector, reuse auth, basic JSM ticket pulling)?
   - Wait for feedback on the recent PRs first?

Happy to move fast once I understand the blocker. Thanks!

**Comment by minnhnh** [comment]
/attempt #2281\n\nPlan: land minimal JSM connector (reuse JiraConnector) + tests + UI wiring; will attach demo video in PR.

**Comment by OsirisLord** [comment]
/attempt #2281

Created PR #11415 implementing the Jira Service Management connector.

The implementation:
- Adds `JIRA_SERVICE_MANAGEMENT` to `DocumentSource` enum (with description - fixing the bug that caused previous PR #7877 to be rejected)
- Creates `JiraServiceManagementConnector` in a new `jira_service_management/` module that reuses existing Jira utilities
- Adds JSM-specific JQL generation with project key filtering and optional issue type filtering
- Full frontend integration: sources.ts, types.ts, connectors.tsx with JSM-specific configuration form
- Supports both Cloud (v3 API) and Server/Data Center (v2 API) with auto-detection
- Supports scoped tokens, comment email blacklist, labels to skip, and JSM issue type filtering
- No Alembic migration needed (source_type column is already sa.String)

PR: https://github.com/onyx-dot-app/onyx/pull/11415

**Comment by minnhnh** [comment]
Opened PR https://github.com/onyx-dot-app/onyx/pull/11434 implementing the Jira Service Management connector. Please review—happy to iterate on any requested changes. (Closes #2281)

**Comment by liyanzong** [comment]
/attempt #2281

I will implement the Jira Service Management Connector.

**Comment by Raibod92** [comment]
/attempt

I'd like to work on this issue (Jira Service Management Connector).

**My approach:**
1. I will study the existing connector structure in the README.
2. I will implement the logic to fetch tickets from Jira Service Management.
3. I will write tests and ensure it passes the CI.

**Estimated time:** just 1 Days.

Let me know if that works.

**Comment by midostes** [comment]
/attempt #2281

Implementing a Jira Service Management connector that delegates to the existing Jira connector (indexes a JSM service desk project, with JSM-specific source tagging + service-desk validation). PR opened: #11722

**Comment by Kaustubhh-Deshmukh-Git** [comment]
/attempt #2281

Plan: Implement Jira Service Management connector by 
extending existing Jira connector. Will reuse auth 
utilities, add JSM source enum, implement ticket 
fetching via /rest/servicedeskapi/ endpoints, wire 
frontend config, add tests and demo video.

**Comment by 64johnlee** [comment]
[PR #11269](https://github.com/onyx-dot-app/onyx/pull/11269) implements this and is ready for review.

It adds a `JiraServiceManagementConnector` that subclasses `JiraConnector` (inheriting pagination, checkpointing, and permission sync), stamps `DocumentSource.JIRA_SERVICE_MANAGEMENT` on every document, and auto-scopes indexing to Service Management (service-desk) projects when no project key / custom JQL is given. Includes the full frontend wiring (source metadata, connector config UI, credential template, auto-sync) and unit tests.

Status: 14/14 unit tests passing, and the JSM-specific behavior (project type key `service_desk`, JQL scoping) has been cross-checked against the live Jira API. Ready for maintainer review whenever convenient.

**Comment by Shaidyk** [comment]
/claim #2281

PR: https://github.com/onyx-dot-app/onyx/pull/12084

**Comment by wywysurf** [comment]
/attempt #2281

Plan: add a `jira_service_management` connector. Since JSM requests are Jira
issues in a service-desk project, I'd reuse the existing Jira connector's proven
building blocks (`build_jira_client`, `_perform_jql_search`, `process_jira_issue`)
rather than duplicate them, targeting the configured JSM project via
`project = "<KEY>" AND updated >= start AND updated <= end`.

Scope:
- backend: new `onyx/connectors/jira_service_management/` (Load + Poll connector),
  `JIRA_SERVICE_MANAGEMENT` added to DocumentSource, factory mapping
- frontend: SOURCE_METADATA_MAP + connectorConfigs entry
- tests under `backend/tests/daily/connectors/`, plus a demo video in the PR

Two clarifications before I build, to avoid rework:
1. Do you want the thin JQL pull of project issues, or full `servicedeskapi`
   enrichment (request types, SLAs, approvals, organizations)?
2. Load+Poll connector, or a CheckpointedConnector to match the Jira one?
3.

**Comment by Delorne007** [comment]
/attempt #2281

Implementation Plan:
I will implement a Jira Service Management (JSM) connector by reusing the existing Jira connector's core modules (build_jira_client, _perform_jql_search, process_jira_issue).

Scope:

Create backend/onyx/connectors/jira_service_management/ for Load+Poll functionality.

Add JIRA_SERVICE_MANAGEMENT to DocumentSource and configure the factory mapping.

Update SOURCE_METADATA_MAP and connectorConfigs in the frontend.

Note on Environment:
I have successfully configured the local environment using Python 3.14.6. During setup, I identified a dependency resolution issue with litellm==1.85.1 (which is unavailable for this Python version), but the connector code is prepared and ready for implementation.

**Comment by NAEL592** [comment]
I’m withdrawing my `/attempt #2281`.

I’m no longer working on this issue, and I don’t have a PR or claim pending for it. Please consider my attempt inactive so others can proceed without waiting on me. Thanks.

**Comment by stmr** [comment]
I can take this if still available/payable.

Planned scope: Jira Service Management Connector.

I will keep the PR/submission focused, include tests/docs where requested, run project checks, and wait for maintainer or bounty-platform confirmation before starting work.

**Comment by bnuckols13** [comment]
Interested in picking up the JSM connector. It looks like #10312 got most of the way there but has gone stale. Before I put in a fresh implementation with integration tests and docs, could a maintainer confirm whether #10312 is still in play, and that a clean, complete PR for this would get a review? Happy to start from scratch following the existing Jira connector. /attempt

**Comment by Delorne007** [comment]
Hi everyone, I am finalizing a clean, production-ready implementation for the Jira Service Management connector (#2281). My plan focuses on:

1)Leveraging existing Jira connector building blocks to ensure consistency and maintainability.

2)Including unit and integration tests to meet the repository's contribution standards.

3)Providing a clear setup guide for API token configuration.

4)Submitting with a full demo video as per bounty requirements.

I have reviewed the stale PRs and believe this approach addresses the previous feedback. I’m ready to push this PR immediately—could a maintainer confirm if they are currently looking for a fresh submission? 
Thank you!

**Comment by ForeverTools** [comment]
/attempt #2281

Implementation plan:
- Subclass `JiraConnector` to inherit all existing Jira authentication, credential management, and checkpoint logic
- Override `_get_jql_query` to scope queries to JSM issue types (Service Request, Incident, Problem, Change, Service Task) unless the user supplies a custom JQL
- Enrich documents with JSM-specific metadata from the Service Desk API (`/rest/servicedeskapi/request/{issueKey}`): request type, SLA breach status, customer display name
- Register `JIRA_SERVICE_MANAGEMENT` in `DocumentSource` and `CONNECTOR_CLASS_MAP`
- Add frontend form config (same fields as Jira)
- Unit tests covering JQL scoping, metadata enrichment, error handling

**Comment by saurav-codes** [comment]
/attempt #2281

Plan: add a Jira Service Management connector by subclassing the existing Jira connector, scoping to service-desk projects via JQL when no explicit project/filter is given, and stamping documents as JIRA_SERVICE_MANAGEMENT. Will wire backend factory, frontend source metadata and form, plus tests.

**Comment by Nabeel-javed** [comment]
Hi @Weves — is there any remaining funded scope for a new contributor under this Jira Service Management connector issue? I see several open implementations, including #11843, #14532 and #14537, so I’d like to clarify what still needs work before preparing another change.

If a focused integration, testing or compatibility gap remains, could you identify the preferred base PR, acceptance criteria and available reward? Is the advertised $250 still available, and would AI-assisted development with reproducible validation be eligible? Please also confirm the current payout route and whether a test JSM environment is supplied; I would need a route that doesn’t require purchasing access. If the funded scope is already covered, no new implementation is needed from me.

**Comment by Ahavahdev1** [comment]
🤖 **MEA Sovereign Agent** assumiu a análise desta issue #2281. Sintetizando a solução...

**Comment by Ahavahdev1** [comment]
✅ A solução para a issue #2281 foi implementada com sucesso no Pull Request: https://github.com/onyx-dot-app/onyx/pull/14622

**Comment by zbzbdzb** [comment]
/attempt #2281

I’m working on this with Codex. The scope I’m targeting is one explicitly selected JSM project, with project-type validation before both indexing and pruning. A failed project lookup should stop the run instead of falling back to an unrestricted Jira query. I’ll reuse the current Jira parsing and checkpointing, keep document IDs consistent between full and slim retrieval, and check that the JSM source stays separate when ordinary Jira is also connected.

I’ve reviewed the current connector guide and the existing implementations. The required live JSM test and UI recording are still to be done; I’m not claiming a completed or tested connector yet. Is a contributor test project available for those checks?

One payment detail to confirm: the linked Algora payout page now returns 404, and its repository documentation describes Stripe Connect but does not list PayPal. If the contribution is accepted, is PayPal an available way to receive the advertised $250 bounty?

**Comment by Furox-Art** [comment]
/attempt #2281

Plan: Implement a Jira Service Management connector following backend/onyx/connectors README conventions — document indexing for a specified JSM project with proper credential handling, incremental sync, and tests, matching the structure of the existing Jira/Confluence connectors.

**Comment by TheAliphant** [comment]
I would like to work on this bounty. Plan: implement a project-scoped Jira Service Management connector following the connector README and existing Jira connector patterns; wire the backend/frontend source registration and credentials; add focused unit coverage for pagination, project validation, and document mapping; run the relevant tests and include the requested demo evidence in the PR. I will keep the change scoped to the stated JSM connector requirements and use `/claim #2281` only in the submitted PR.

**Comment by Furox-Art** [comment]
Status update on this connector — there is now a complete, tested implementation in PR #14687:

- **Backend**: dedicated `jira_service_management` connector (connector + utils + registry wiring), following the same patterns as the existing Jira connector
- **Web UI**: connector entry, credential schema, source registration
- **Tests**: unit suite for the connector + utils, plus a daily integration test (`test_jsm_basic.py`)

The earlier claimed PRs on this issue (#3087, #7877) are closed, so this should currently be the only open implementation. The branch CI is waiting on first-contribution workflow approval (`action_required`) — a maintainer approve click would let the checks actually run. Happy to adjust anything on review.

**Comment by stffinfcti** [comment]
/attempt #2281

Plan: dedicated Jira Service Management connector following `backend/onyx/connectors/README.md` — subclass the existing Jira connector machinery (checkpointed JQL pagination, slim docs, perm sync), stamp `DocumentSource.JIRA_SERVICE_MANAGEMENT`, scope indexing to a specified service-desk project, enrich docs with JSM-specific metadata (request type, organizations, SLA status, internal-note filtering), add `include_attachments` parity for full + slim passes, wire backend registry/constants/Slack icon + frontend source/config/credentials, plus unit tests with a mocked Jira API and a daily connector test.

**Comment by KazTradez** [comment]
/attempt #2281

Plan: implement a Jira Service Management connector following backend/onyx/connectors/README.md conventions:

- New `JiraServiceManagementConnector` subclassing the existing `JiraConnector` — JSM shares Jira's REST surface (issues/JQL/credentials), so checkpointed pagination, slim-doc retrieval, permission sync, and document processing are reused rather than duplicated.
- New `DocumentSource.JIRA_SERVICE_MANAGEMENT` registered in the connector registry, with one upstream fix: `process_jira_issue` currently hardcodes `source=DocumentSource.JIRA`, so it needs an optional `source` parameter for subclasses to label their documents correctly.
- Frontend contract: SOURCE_METADATA_MAP entry, connector form config (base URL / scoped token / project-or-JQL tabs), credential JSON shape (same as Jira), and Slack bot icon mapping.
- Tests under `backend/tests/daily/connectors/jira_service_management/` mirroring the existing Jira connector tests (validation, basic indexing, scoped token, slim docs).

Will open a draft PR early for visibility. Happy to adjust scope if maintainers prefer a different approach (e.g. JSM-specific field handling for service requests).

**Comment by xtuit** [comment]
Before starting any implementation, could a maintainer confirm whether the $250 bounty for this issue is still available, whether new submissions are still being reviewed, and whether the acceptance requirements have changed since this issue was opened? I have not started work or claimed the bounty.
